### PR TITLE
add DateTime class

### PR DESCRIPTION
**Pull request by drdzyk**
This PR adds some functionality of php [DateTime](https://www.php.net/manual/en/class.datetime.php) class. Namelly, the following functions:

```

class DateTime {
  /** @kphp-extern-func-info can_throw */
  public function __construct(string $datetime = "now", ?DateTimeZone $timezone = null);
  public static function createFromFormat(string $format, string $datetime, ?DateTimeZone $timezone = null): ?DateTime;
  public static function getLastErrors(): array|false;
  public function modify(string $modifier): ?DateTime;
  public function setDate(int $year, int $month, int $day): DateTime;
  public function setISODate(int $year, int $week, int $dayOfWeek = 1): DateTime;
  public function setTime(
      int $hour,
      int $minute,
      int $second = 0,
      int $microsecond = 0
  ): DateTime;
  public function setTimestamp(int $timestamp): DateTime;
  public function format(string $format): string;
  public function getOffset(): int;
  public function getTimestamp(): int;
}
```

### CODE REVIEW

File: runtime/timelib_wrapper.cpp [hunk 1]
```diff
@@ -215,3 +225,511 @@ std::pair<int64_t, bool> php_timelib_strtotime(const string &tz_name, const stri
...
+std::pair<timelib_time *, string> php_timelib_date_initialize(const string &tz_name, const string &time_str, const char *format) {
+  auto script_guard = make_malloc_replacement_with_script_allocator();
+
+  const string &time_str_new = time_str.empty() ? NOW : time_str;
+  timelib_error_container *err = nullptr;
+  timelib_time *t = format
+    ? timelib_parse_from_format(format, time_str.c_str(), time_str.size(), &err, timelib_builtin_db(), timelib_parse_tzfile)
+    : timelib_strtotime(time_str_new.c_str(), time_str_new.size(), &err, timelib_builtin_db(), timelib_parse_tzfile);
+
+  /* update last errors and warnings */
```
Comment (quasilyte) [inline runtime/timelib_wrapper.cpp:267]: I'm not sure it's common to use `/**/` style  comments in our codebase (except for some top-level doc comments).

File: runtime/timelib_wrapper.cpp [hunk 2]
```diff
@@ -215,3 +225,511 @@ std::pair<int64_t, bool> php_timelib_strtotime(const string &tz_name, const stri
...
+  if (err && err->error_count) {
+    /* spit out the first library error message, at least */
+    return {false, gen_parse_error_msg(*err, modifier)};
+  }
+
+  std::memcpy(&t->relative, &tmp_time->relative, sizeof(timelib_rel_time));
+  t->have_relative = tmp_time->have_relative;
+  t->sse_uptodate = 0;
+
+  if (tmp_time->y != -99999) {
```
Comment (quasilyte) [inline runtime/timelib_wrapper.cpp:651]: Maybe add a named const for `-99999`?
It's declared in timelib as follows:
```
#define TIMELIB_UNSET   -99999
```

File: runtime/timelib_wrapper.cpp (hunk 1 above)
Comment (drdzyk) [reply runtime/timelib_wrapper.cpp]: I prefer `//` comments too, `/**/` appears just from copy-paste, fixed.

File: runtime/timelib_wrapper.cpp (hunk 2 above)
Comment (drdzyk) [reply runtime/timelib_wrapper.cpp]: fixed

**Review by quasilyte — APPROVED** [review]

**Pull request merged**